stac_fastapi/pgstac/stac_fastapi/pgstac/core.py

```diff
@@ -38,13 +37,11 @@
 class CoreCrudClient(AsyncBaseCoreClient):
     """Client for core endpoints defined by stac."""
 
-    async def all_collections(self, **kwargs) -> Collections:
+    async def all_collections(self, request: Request, **kwargs) -> Collections:
         """Read all collections from the database."""
-        request: Request = kwargs["request"]
         base_url = get_base_url(request)
-        pool = request.app.state.readpool
 
-        async with pool.acquire() as conn:
+        async with request.app.state.get_connection(request, "r") as conn:
```

Comment:
To be honest it feels kinda weird to have to pass `request` to an function we get from the request object but sadly I don't think we have any choice 😭